fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonPredicateConverter.java

```diff
@@ -114,10 +133,13 @@ public Predicate visitBinaryPredicate(BinaryPredicateOperator operator, Void con
             return null;
         }
         int idx = fieldNames.indexOf(columnName);
-        Object literal = getLiteral(operator.getChild(1));
+        Object literal = getLiteral(operator.getChild(1), fieldTypes.get(idx));
         if (literal == null) {
             return null;
         }
+        if (fieldTypes.get(idx) instanceof BooleanType) {
+            literal = convertBoolLiteralValue(literal);
```

Comment:
why only check this type?

Reply:
As commented below, it's for supporting both 0/1 and true/false for boolean type.